src/App.tsx

```diff
@@ -1,29 +1,61 @@
 import React, { Component } from 'react';
-// @ts-ignore
+import {BrowserRouter as Router, Route, Link } from 'react-router-dom';
 import logo from './logo.svg';
 import './App.css';
 
-class App extends Component {
+class AppRouter extends Component {
+  render() {
+    return (
+      <Router>
+        <div>
+          <Route path='/' exact component={HomePage} />
+          <Route path='/login' component={LoginPage} />
+        </div>
+      </Router>
+    );
+  }
+}
+
+interface User {
+  email: string;
+}
+
+class HomePage extends Component {
+  readonly state: {user: User | null} = {user: null};
+
+  componentDidMount() {
```

Comment:
😮 we don't have to convert this (it's totally fine as is) but if we want to use react hooks we can here and it'll be so much fun 😄

Reply:
What will react hooks do?